### DISCUSSION TITLE
Exporting the types in package.json

### DISCUSSION THREAD

**Discussion by Lemour-sudo** [cloudscape-design/components#2157] [Q&A] [answered]
Shouldn't the types also be exported in the package.json file for each export: https://github.com/cloudscape-design/code-view/blob/main/package.json#L37

This can avoid import errors for different module resolutions: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing

**Comment by abdhalees** [comment, marked as answer]
types exports is needed when `.d.ts` file is in a different location than the exported `js` file, in our source code all type files are in the same location as the js files therefore this shouldn't be needed.

Did you find any import where this caused you issues?

**Comment by Lemour-sudo** [comment]
Thanks for the reply.
I am experiencing build problems when building for the "node" moduleResolution. Looks like it does not work for packages with the exports in the package.json.

**Reply by just-boris** [reply]
Make sure you are using the latest typescript and `moduleResolution: "bundler"`. This setup definitely gives the best dev experiecne
